target/linux/ramips/dts/mt7620n_netgear_jwnr2010-v5.dts

```diff
@@ -0,0 +1,137 @@
+/dts-v1/;
+
+#include "mt7620n.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "netgear,jwnr2010-v5", "ralink,mt7620n-soc";
+	model = "Netgear JWNR2010 v5";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			label = "jwnr2010-v5:green:power";
+			gpios = <&gpio0 21 GPIO_ACTIVE_LOW>;
+		};
+
+		wan {
+			label = "jwnr2010-v5:green:wan";
+			gpios = <&gpio0 19 GPIO_ACTIVE_HIGH>;
+		};
+
+		wlan {
+			label = "jwnr2010-v5:green:wlan";
+			gpios = <&gpio3 0 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0radio";
+		};
+
+		lan1 {
+			label = "jwnr2010-v5:green:lan1";
+			gpios = <&gpio2 3 GPIO_ACTIVE_LOW>;
+		};
+
+		lan2 {
+			label = "jwnr2010-v5:green:lan2";
+			gpios = <&gpio2 2 GPIO_ACTIVE_LOW>;
+		};
+
+		lan3 {
+			label = "jwnr2010-v5:green:lan3";
+			gpios = <&gpio2 1 GPIO_ACTIVE_LOW>;
+		};
+
+		lan4 {
+			label = "jwnr2010-v5:green:lan4";
+			gpios = <&gpio2 0 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 20 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+};
+
+&gpio0 {
+	status = "okay";
+};
+
+&gpio2 {
+	status = "okay";
+};
+
+&gpio3 {
+	status = "okay";
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
```

Comment:
Can this go faster?

Reply:
I checked other mt7620n targets, all r 10000000. What r u suggesting ? 12500000 @adrianschmutzler

Reply:
10MHz is used as default because everybody just copies it from some other file. Most flashs support higher frequencies which will increase read speed by a factor 2-3.

Typical approach is to check the flashs datasheet what would be the maximum, and then check with different frequencies set and test read speed on the device. (Make sure you have a recovery mechanism in place, as choosing a too high frequency will prevent boot.)

Reply:
> 10MHz is used as default because everybody just copies it from some other file. Most flashs support higher frequencies which will increase read speed by a factor 2-3.
> 
> Typical approach is to check the flashs datasheet what would be the maximum, and then check with different frequencies set and test read speed on the device. (Make sure you have a recovery mechanism in place, as choosing a too high frequency will prevent boot.)

Hi, I checked the datasheet. It says something like 33MHz (Normal Read) / 86MHz (Fast Read) @adrianschmutzler

Reply:
30MHz seems working for me, from datasheet it says upto 33MHz..I saw some commit where mt7620 getting 30MHz bump..so keeping it 30MHz for now. @adrianschmutzler

Reply:
30 or 33 MHz might be a good compromise.